src/cart/crt_hg_proc.c

```diff
@@ -626,11 +626,21 @@ crt_proc_in_common(crt_proc_t proc, crt_rpc_input_t *data)
 			hdr = &rpc_priv->crp_req_hdr;
 
 			hdr->cch_flags = rpc_priv->crp_flags;
-			hdr->cch_rank = crt_grp_priv_get_primary_rank(
+			hdr->cch_dst_rank = crt_grp_priv_get_primary_rank(
 						rpc_priv->crp_grp_priv,
 						rpc_priv->crp_pub.cr_ep.ep_rank
 						);
-			hdr->cch_tag = rpc_priv->crp_pub.cr_ep.ep_tag;
+			hdr->cch_dst_tag = rpc_priv->crp_pub.cr_ep.ep_tag;
+
+			if (crt_is_service())
+				hdr->cch_src_rank =
+					crt_grp_priv_get_primary_rank(
+						rpc_priv->crp_grp_priv,
+						rpc_priv->crp_grp_priv->gp_self
+						);
+			else
+				hdr->cch_src_rank = CRT_NO_RANK;
```

Comment:
This is a question for my understanding only, no change requested:
This is the RPC issuer / encode side. When the RPC is received by the "cch_dst_rank" node, it will see all of these relevant fields (cch_dst_rank, cch_dst_tag, cch_src_rank) by virtue of crt_hg_unpack_header()->crt_proc_common_hdr() decode -> hg_proc_memcpy()? i.e,, there is no other "decode" side logic that needs to specifically assign these fields for the recipient of the RPC?

Reply:
Correct, when the header is decoded in the recipient side, it will take care of it